geo-types/src/coordinate.rs

```diff
@@ -25,12 +27,45 @@ use approx::{AbsDiffEq, RelativeEq, UlpsEq};
 /// [vector space]: //en.wikipedia.org/wiki/Vector_space
 #[derive(Eq, PartialEq, Clone, Copy, Debug, Hash, Default)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct Coordinate<T: CoordNum> {
+pub struct Coordinate<T: CoordNum, Z: CoordNum = NoValue, M: CoordNum = NoValue> {
     pub x: T,
     pub y: T,
+    pub z: Z,
+    pub m: M,
 }
 
+impl<T: CoordNum, Z: CoordNum, M: CoordNum> Coordinate<T, Z, M> {
+    /// Create a new instance of a coordinate.
+    /// **ATTENTION** Use [`coord!`] macro or one of the [`from`] methods instead.
+    /// This method exists because it seems instantiation is slower from a macro
+    /// without a dedicated method. This fn should go away once perf issues are resolved.
+    #[inline]
+    #[doc(hidden)]
+    pub fn new__(x: T, y: T, z: Z, m: M) -> Self {
+        Self { x, y, z, m }
+    }
+}
+
+/// A lightweight struct used to store coordinates on the 2-dimensional
+/// Cartesian plane together with a Measure value of the same type.
+///
+/// See also [`Coordinate`]
+pub type CoordinateM<T, M = T> = Coordinate<T, NoValue, M>;
+
+/// A lightweight struct used to store coordinates on the 3-dimensional
+/// Cartesian plane.
+///
+/// See also [`Coordinate`]
+pub type Coordinate3D<T> = Coordinate<T, T, NoValue>;
+
+/// A lightweight struct used to store coordinates on the 3-dimensional
+/// Cartesian plane together with a Measure value of the same type.
+///
+/// See also [`Coordinate`]
+pub type Coordinate3DM<T, M = T> = Coordinate<T, T, M>;
+
 impl<T: CoordNum> From<(T, T)> for Coordinate<T> {
```

Comment:
While working on #816 - I realized it probably doesn't make sense to implement this for all Coordinate<T>, rather only for Coordinate<T, NoValue, NoValue>. And same for `From([T; 2])` below and the similar impls in `Point`. 

We probably should eventually have a way to easily project 3d+ points to 2d, but it seems like it should be more explicit.

WDYT?